src/Library/public/usbguard/Rule.cpp

```diff
@@ -243,14 +245,20 @@ namespace usbguard {
   Rule::operator bool() const
   {
     return !(getTarget() == Target::Unknown ||
-	     getTarget() == Target::Invalid);
+        getTarget() == Target::Invalid);
   }
 
   std::string Rule::toString(bool invalid) const
   {
     return d_pointer->toString(invalid);
   }
 
+  std::string Rule::toLDIF(std::string hostname, std::string name_prefix, std::string USBGUARD_BASE, std::string objclass,
```

Comment:
I don't like that you are introducing LDAP specifics into the Rule interface. This has to go away.

And btw. why passing by value and not reference?